test/functional/ios/webdriver_tests.py

```diff
@@ -24,20 +27,31 @@ class WebDriverTests(unittest.TestCase):
     def setUp(self):
         desired_caps = desired_capabilities.get_desired_capabilities('UICatalog.app.zip')
         self.driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)
-        desired_caps['deviceName'] = 'iPhone Xs Max'
-        desired_caps['wdaLocalPort'] = '8102'
 
     def tearDown(self):
         self.driver.quit()
 
-    def testAllSessions(self):
+    def test_all_sessions(self):
+        port = get_available_port_range(8200, 8300)
         desired_caps = desired_capabilities.get_desired_capabilities('UICatalog.app.zip')
         desired_caps['deviceName'] = 'iPhone Xs Max'
-        desired_caps['wdaLocalPort'] = '8102'
-        self.driver1 = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)
-        self.assertEqual(2, len(self.driver.all_sessions))
-        if self.driver1:
-            self.driver1.quit()
+        desired_caps['wdaLocalPort'] = port
+
+        class get_all_sessions(object):
```

Comment:
This class name doesn't mean actual behavior and it's a little tricky.
I'd rather have your original code style from maintenance point of view if it works.

```
WebDriverWait(driver2, 10).until(len(driver.all_sessions) == 2)

or

TIMEOUT = 10
WebDriverWait(driver2, TIMEOUT).until(len(driver.all_sessions) == 2)
```

Reply:
`len(driver.all_sessions) == 2` did not work in `WebDriverWait`, unfortunately